Sources/MongoSwift/MongoCollection.swift

```diff
@@ -0,0 +1,64 @@
+import libmongoc
+
+/// A MongoDB collection.
+public class MongoCollection<T: Codable> {
+    internal var _collection: OpaquePointer?
+    internal var _client: MongoClient?
+
+    /// A `Codable` type associated with this `MongoCollection` instance. 
+    /// This allows `CollectionType` values to be directly inserted into and 
+    /// retrieved from the collection, by encoding/decoding them using the 
+    /// `BsonEncoder` and `BsonDecoder`. 
+    /// This type association only exists in the context of this particular 
+    /// `MongoCollection` instance. It is the responsibility of the user to 
+    /// ensure that any data already stored in the collection was encoded 
+    /// from this same type.
+    public typealias CollectionType = T
+
+    /// The name of this collection.
+    public var name: String {
+        return String(cString: mongoc_collection_get_name(self._collection))
+    }
+
+    /// The `ReadConcern` set on this collection, or `nil` if one is not set.
+    public var readConcern: ReadConcern? {
+        // per libmongoc docs, we don't need to handle freeing this ourselves
+        let readConcern = mongoc_collection_get_read_concern(self._collection)
+        let rcObj = ReadConcern(from: readConcern)
+        if rcObj.isDefault { return nil }
```

Comment:
Would it make sense to use a single `return` with a ternary for this kind of thing? Totally unrelated to the PR -- just a general coding style question.